shopping-cart/shopping-cart-scala/shopping-cart/src/main/scala/com/example/shoppingcart/impl/ShoppingCartReportRepository.scala

```diff
@@ -0,0 +1,58 @@
+package com.example.shoppingcart.impl
+
+import java.time.OffsetDateTime
+
+import akka.Done
+import com.example.shoppingcart.api.ShoppingCartReport
+import slick.dbio.DBIO
+import slick.jdbc.PostgresProfile.api._
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+/**
+ * The report repository manage the storage of ShoppingCartReport which is a API class (view model).
+ *
+ * It saves data in a ready for consumption format for that specific API model.
+ * If the API changes, we must regenerated the stored models.
+ */
+class ShoppingCartReportRepository(database: Database) {
+
+  class ShoppingCartReportTable(tag: Tag) extends Table[ShoppingCartReport](tag, "shopping_cart_report") {
+    def cartId = column[String]("cart_id", O.PrimaryKey)
+
+    def creationDate = column[OffsetDateTime]("creation_date")
+
+    def checkoutDate = column[Option[OffsetDateTime]]("checkout_date")
+
+    def * = (cartId, creationDate, checkoutDate) <> ((ShoppingCartReport.apply _).tupled, ShoppingCartReport.unapply)
+  }
+
+  val reportTable = TableQuery[ShoppingCartReportTable]
+
+  def createTable() = reportTable.schema.createIfNotExists
+
+  def findById(id: String): Future[Option[ShoppingCartReport]] =
+    database.run(findByIdQuery(id))
+
+  def updateCart(cartId: String, qty: Int, dateTime: OffsetDateTime): DBIO[Done] = {
```

Comment:
`qty` is never used here. Should it be part of `ShoppingCartReport`?

Reply:
Also, what does `dateTime` represent here?

Reply:
`qty` is a leftover. I will remove.